src/runtime_src/core/tools/common/ReportAieShim.cpp

```diff
@@ -256,13 +258,22 @@ ReportAieShim::getPropertyTree20202(const xrt_core::device * _pDevice,
 void 
 ReportAieShim::writeReport( const xrt_core::device* /*_pDevice*/,
                             const boost::property_tree::ptree& _pt, 
-                            const std::vector<std::string>& /*_elementsFilter*/,
+                            const std::vector<std::string>& _elementsFilter,
                             std::ostream & _output) const
 {
   boost::property_tree::ptree empty_ptree;
+  std::vector<std::string> aieTileList;
 
   _output << "AIE\n";
 
+  for (auto it = _elementsFilter.begin(); it != _elementsFilter.end(); ++it) {
+    if(*it == "tiles") {
+      auto tile_list = std::next(it);
+      if (tile_list != _elementsFilter.end())
+        boost::split(aieTileList, *tile_list, boost::is_any_of(","));
+    }
+  }
+

```

Comment:
This looks like déjà vu.
Should we have some common functions instead to process the `_elementsFilter` and pass some actions as lambdas?

Reply:
_elementsFilter  is not used very commonly. Adding command function will lead more code change.